### PR TITLE
Porting l10n_it_ddt to 9.0

### PR DESCRIPTION
**Pull request by eLBati**
Depends on https://github.com/OCA/stock-logistics-workflow/pull/199

### CODE REVIEW

File: l10n_it_ddt/__openerp__.py [hunk 1]
```diff
@@ -0,0 +1,40 @@
...
+#    Copyright (C) 2015 Apulia Software s.r.l. (http://www.apuliasoftware.it)
+#    @author Francesco Apruzzese <f.apruzzese@apuliasoftware.it>
+#
+#    License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl.html).
+#
+##############################################################################
+
+{
+    'name': 'DDT',
+    'version': '8.0.1.0.0',
```
Comment (gigidn) [inline l10n_it_ddt/__openerp__.py]: upgrade to 9.0.X.Y.Z

File: l10n_it_ddt/models/account.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+from openerp import fields, models, api
+
+
+class AccountInvoice(models.Model):
+
+    _inherit = 'account.invoice'
+
+    carriage_condition_id = fields.Many2one(
+        'stock.picking.carriage_condition', 'Carriage Condition')
```
Comment (gigidn) [inline l10n_it_ddt/models/account.py]: should be better use string='Carriage Condition'?

File: l10n_it_ddt/models/stock.py [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+    @api.multi
+    def write(self, values):
+        if 'move_lines' in values:
+            pack_to_update = self.env['stock.picking.package.preparation']
+            for picking in self:
+                pack_to_update |= picking.ddt_ids
+        res = super(StockPicking, self).write(values)
+        if 'move_lines' in values:
+            if pack_to_update:
+                pack_to_update._update_line_ids()
```
Comment (gigidn) [inline l10n_it_ddt/models/stock.py]: better for code performance check only 1 'move_lines' in values

Comment (OpenCode) [inline l10n_it_ddt/models/stock.py]: Why?

Comment (eLBati) [inline l10n_it_ddt/models/stock.py]: @gigidn could you show how you would change this?

**Comment by eLBati** [comment]
@gigidn Thanks for review.
I addressed your remarks

File: l10n_it_ddt/models/stock.py (hunk 3 above)
Comment (gigidn) [inline l10n_it_ddt/models/stock.py]: One way
    @api.multi
    def write(self, values):
        pack_to_update = None
        if 'move_lines' in values:
            pack_to_update = self.env['stock.picking.package.preparation']
            for picking in self:
                pack_to_update |= picking.ddt_ids
        res = super(StockPicking, self).write(values)
        if pack_to_update:
            pack_to_update._update_line_ids()
        return res

or this better if super call is not influenced by pack_to_update._update_line_ids() call
    @api.multi
    def write(self, values):
        if 'move_lines' in values:
            pack_to_update = self.env['stock.picking.package.preparation']
            for picking in self:
                pack_to_update |= picking.ddt_ids
            pack_to_update._update_line_ids()
        return super(StockPicking, self).write(values)

**Comment by guastallidavide** [comment]
Hi, how can I test it if stock_picking_package_preparation is not yet ported ?
Am I missing something?

**Comment by eLBati** [comment]
@gigidn improved write method with https://github.com/OCA/l10n-italy/pull/184/commits/767d2f1801f61c07c89c971489716110ec139de3

**Comment by tafaRU** [comment]
@eLBati, some travis builds are red, please check them.

**Comment by eLBati** [comment]
@tafaRU travis is red because this depends on https://github.com/OCA/stock-logistics-workflow/pull/199

Fixed flake8 with https://github.com/OCA/l10n-italy/pull/184/commits/e3861417f6fabbb1fcb1de721f71008e12711912

**Comment by eLBati** [comment]
@OpenCode @CiroBoxHub @andrea4ever @archetipo coverage should reach 100% now

**Comment by OpenCode** [comment]
@eLBati you're crazy :D

File: l10n_it_ddt/wizard/ddt_from_picking.py [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+        transportation_method_id = False
+        for picking in self.picking_ids:
+            if picking.sale_id and (
+                    picking.sale_id.transportation_method_id):
+                if transportation_method_id and (
+                    transportation_method_id != (
+                        picking.sale_id.transportation_method_id)):
+                    raise UserError(
+                        _("Selected Pickings have"
+                          " different transportation reason"))
```
Comment (espo-tony) [inline l10n_it_ddt/wizard/ddt_from_picking.py]: The error message is wrong: it should rather be "different transportation method"

**Comment by eLBati** [comment]
@espo-tony thanks, done

**Comment by espo-tony** [comment]
:+1:

**Comment by scigghia** [comment]
:+1:

**Pull request merged**